server/webapp/WEB-INF/rails/webpack/models/pipeline_configs/job.ts

```diff
@@ -15,31 +15,41 @@
  */
 
 import JsonUtils from "helpers/json_utils";
-import {Stream} from "mithril/stream";
+import * as _ from "lodash";
 import * as stream from "mithril/stream";
+import {Stream} from "mithril/stream";
 import {ValidatableMixin} from "models/mixins/new_validatable_mixin";
+import {EnvironmentVariableConfig} from "models/pipeline_configs/environment_variable_config";
 import {Task} from "models/pipeline_configs/task";
+import * as s from "underscore.string";
 
 export class Job extends ValidatableMixin {
   name: Stream<string>;
+  environmentVariables: Stream<EnvironmentVariableConfig[]>;
   tasks: Stream<Task[]>;
 
-  constructor(name: string, tasks: Task[]) {
+  constructor(name: string, tasks: Task[], envVars: EnvironmentVariableConfig[]) {
```

Comment:
Make optional? i.e., `envVars?: EnvironmentVariableConfig[]`. Perhaps consider storing as `NameableSet` since these should be unique by name.